TechliftTelegramBot/Services/BotCommandCheckService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using TechliftTelegramBot.Models;
+using Telegram.Bot;
+using Telegram.Bot.Types;
+
+namespace TechliftTelegramBot.Services
+{
+    public class BotCommandCheckService : IBotCommandCheckService
+    {
+        private readonly ILogger _logger;
+        private readonly ApplicationConfiguration _config;
+        private readonly TelegramBotClient _botclient;
+        private IEnumerable<BotCommand> CommandsToBeSet;
+        public IEnumerable<BotCommand> AllCommands = new BotCommand[]
+            {
+                new BotCommand{Command="todays_profit",Description="get today's profit"},
+                new BotCommand{Command="remaining_limit",Description="get remaining limit"},
+                new BotCommand{Command="set_limit",Description="set limit of a user"},
+                new BotCommand{Command="week_profit",Description="get weekly profit"},
+            };
+       
+        public BotCommandCheckService(ILogger<BotCommandCheckService> logger, IOptions<ApplicationConfiguration> config)
+        {
+            _logger = logger;
+            _config = config.Value;
+            _botclient =new TelegramBotClient(_config.APIToken);
+        }
+
+        public async void CheckCommands()
+        {
+            CommandsToBeSet = await _botclient.GetMyCommandsAsync();
+            bool CommandsNeedToReset = false;
+            foreach (BotCommand botCommand in AllCommands)
+            {
+                if (CommandsToBeSet.Any(any => any.Command == botCommand.Command) != true)
+                {
+                    CommandsNeedToReset = true;
```

Comment:
break when first encounter of this, why do you have to continue with the entire loop